client/containers/DashboardSubmissionWorkflow/ExistingSubmissionWorkflowEditor.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import { Button } from '@blueprintjs/core';
+
+import { SubmissionWorkflow } from 'types';
+import { usePageContext } from 'utils/hooks';
+import { usePersistableState } from 'client/utils/usePersistableState';
+
+import { updateSubmissionWorkflow } from './api';
+import SubmissionWorkflowEditor from './SubmissionWorkflowEditor';
+import DashboardSubmissionWorkflowFrame from './DashboardSubmissionWorkflowFrame';
+
+type Props = {
+	initialWorkflow: SubmissionWorkflow;
+	onWorkflowUpdated: (w: SubmissionWorkflow) => unknown;
+};
+
+const ExistingSubmissionWorkflowEditor = (props: Props) => {
+	const { onWorkflowUpdated, initialWorkflow } = props;
+	const [isValid, setIsValid] = useState(false);
+
+	const {
+		scopeData: {
+			elements: { activeCollection },
+		},
+	} = usePageContext();
+
+	const {
+		state: workflow,
+		update: updateWorkflow,
+		persist,
+		hasChanges,
+		isPersisting,
+	} = usePersistableState(initialWorkflow, async (update, full) => {
+		await updateSubmissionWorkflow(update, activeCollection.id);
+		// Give the usePersistableState hook disable its onBeforeUnload before the parent redirects
```

Comment:
I'm not 100% sure what this comment means, could you help clarify it for me?

Reply:
Oh oops that was grammatically incoherent but even then still kind of hard to parse. I wrote a better one.